sch_fq.c

```diff
@@ -479,15 +655,62 @@ static int fq_enqueue(struct sk_buff *skb, struct Qdisc *sch,
 	f->qlen++;
 	qdisc_qstats_backlog_inc(sch, skb);
 	if (fq_flow_is_detached(f)) {
-		fq_flow_add_tail(&q->new_flows, f);
+
+	    fq_flow_add_tail(&q->new_flows, f);
+
+
 		if (time_after(jiffies, f->age + q->flow_refill_delay))
 			f->credit = max_t(u32, f->credit, q->quantum);
 		q->inactive_flows--;
 	}
 
 	/* Note: this overwrites f->age */
+
+        //printk("hash of flow value : %u \n ", (f->socket_hash & q->orphan_mask) );
+
+        /*printk("skb get hash value  : %u \n ", skb_get_hash(skb));
+
+        unsigned long pHash = skb_get_hash(skb) & q->orphan_mask;
+
+        printk("pHash value  : %lu \n ", pHash);*/
+
 	flow_queue_add(f, skb);
 
+	/*
+
+	setting the barrier bits
+	there are 100000 barriers for the co-flow
+
+	individual bits are set by two variabes
+
+	time to send being modified
+
+	*/
+
+	if(f->socket_hash == pFlowid[0])
+	{
+	barrier[barriercounter_flow[0]] = barrier[barriercounter_flow[0]] | 1 << 0;
+	
+	printk("pflow id 0  is   : %d \n ", pFlowid[0]);
+
+	fq_skb_cb(skb)->time_to_send = ktime_get_ns() + 10000000;
+
+        barriercounter_flow[0]++;
+	}
+
+	if(f->socket_hash == pFlowid[1])
+	{
+
+	barrier[barriercounter_flow[1]] = barrier[barriercounter_flow[1]] | 1 << 1;
+	
+	printk("pflow id 1  is   : %d \n ", pFlowid[1]);
+
+	fq_skb_cb(skb)->time_to_send = ktime_get_ns() + 10000000;
+
+	barriercounter_flow[1]++;
```

Comment:
Duplicating code is a bad idea. Please see comment on code duplication above.
I'd refactor this with a loop variable `k` that uses `barriercounter_flow[k]` counter you would use and set the bit by OR'ing with `1 << k`